datacapture/src/main/java/com/google/android/fhir/datacapture/mapping/ResourceMapper.kt

```diff
@@ -607,7 +619,7 @@ private fun Class<*>.getFieldOrNull(name: String): Field? {
   return try {
     getDeclaredField(name)
   } catch (ex: NoSuchFieldException) {
-    return null
+    return superclass.getFieldOrNull(name)
```

Comment:
superclass is nullable and may cause `NullPointerException` incase the field is not found throughout the class hierarchy.
```suggestion
    return superclass?.getFieldOrNull(name)
```